common/authen.js

```diff
@@ -154,6 +158,7 @@
                         // To avoid problems when user explicitly close the modal
                         modalInstance.result.then(onModalClose, onModalClose);
 
+                        // TODO: Attach .catch w/ catchAll
```

Comment:
Based on [this example](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch#Gotchas_when_throwing_errors) in MDN, it was my understanding that if `watchChangeInReferrerId` threw an exception in `$interval`, then it wouldn't be caught by a`.catch` at the end of the outer promise because `$interval` merely schedules an action to be performed. So an extra .catch is needed here in case `watchChangeInReferrerId` throws an exception.

Though, based on earlier discussion today about `$exceptionHandler` on Slack, this catch might not be necessary in the end since `$interval` is Angular code.